libraries/AdaptiveExpressions/Memory/IMemory.cs

```diff
@@ -23,8 +23,9 @@ public interface IMemory
         /// </summary>
         /// <param name="path">memory path.</param>
         /// <param name="value">resolved value.</param>
+        /// <param name="allowSubstitution">allow substitution at null or not.</param>
         /// <returns> true if the memory contains an element with the specified key; otherwise, false.</returns>
-        bool TryGetValue(string path, out object value);
+        bool TryGetValue(string path, out object value, bool allowSubstitution = true);
```

Comment:
This interface has shipped, this is a breaking change.  You will have to do IMemory2

Reply:
If I add a new method into this interface, will that consider a breaking change?

Reply:
We should not introduce this as a breaking change. Adding new method seems fine to me though. @tomlm thoughts?

Reply:
New method is still breaking though--you have to update your interface.  The only option would be IMemory2 or to break,.